File: cpp/1365.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        int n = nums.size();
        int f[101] = {0};
        for (int i : nums)
            ++f[i];
        int ps = 0;
        for (int i = 0; i < 101; i++) {
            ps += f[i];
            f[i] = ps - f[i];
        }
        for (int i = 0; i < n; i++)
            nums[i] = f[nums[i]];
        return nums;
    }
};

File: python3/1365.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        n = len(nums)
        f = [0] * 101
        for i in nums:
            f[i] += 1
        ps = 0
        for i in range(101):
            ps += f[i]
            f[i] = ps - f[i]
        for i in range(n):
            nums[i] = f[nums[i]]
        return nums
